R/target_market_share.R

```diff
@@ -267,14 +267,14 @@ add_percentage_of_initial_production_by_scope <- function(data,
         scope == "sector" ~ percentage_of_initial_sector_production
       )
     ) %>%
-    select(
-      -.data$target_name,
-      -.data$sector_production,
-      -.data$initial_technology_production,
-      -.data$initial_sector_production,
-      -.data$percentage_of_initial_technology_production,
-      -.data$percentage_of_initial_sector_production
-    )
+    select(-any_of(c(
```

Comment:
why `any_of()` here instead of `all_of()`?

Reply:
Think-o

Reply:
🤷 sorry, I still don't get it. I get that `any_of()` is "safer" in the sense that if one of the variables is missing, it will not error out. But `all_of()` seems like a more faithful translation of the original code, no?
``` r
library(dplyr)
df <- tibble(a = 1, b = 2, c = 3)

select(df, -.data$b, -.data$c, -.data$d)
#> Error in `select()`:
#> ! Column `d` not found in `.data`.

select(df, -c(.data$b, .data$c, .data$d))
#> Error in `select()`:
#> ! Column `d` not found in `.data`.

select(df, -all_of(c("b", "c", "d")))
#> Error in `select()`:
#> ! Can't subset columns that don't exist.
#> ✖ Column `d` doesn't exist.

select(df, -any_of(c("b", "c", "d")))
#> # A tibble: 1 × 1
#>       a
#>   <dbl>
#> 1     1
```

You also translated R/summarize_weighted_production.R#L195-199 from...
``` r
select(-c(
  .data$id_loan,
  .data$loan_size_credit_limit,
  .data$loan_size_outstanding
)) %>%
```
to...
``` r
  select(-all_of(c(
    "id_loan",
    "loan_size_credit_limit",
    "loan_size_outstanding"
  ))) %>%
```

Sorry, not trying to be difficult, just trying to understand if there's something I'm missing here. (We're going to have to make similar changes in many other repos.)

Reply:
I meant it's a think-o in that I meant to type `all_of()`, not `any_of()`.

Reply:
great, thanks!